### PR TITLE
Prepare test infrastructure and rename promisify decorator

### PR DESCRIPTION
**Pull request by rosen-vladimirov**
## Test infrastructure
Prepare test infrastructure by:
* Add tests dir and default .d.ts file.
* Modify package.json - add test script.
* Modify .gitignore to exclude test specific files.
* Get yok tests from appbuilder-cli and add some additional ones.
* Add decorators tests.
* Add mocha and chai definition files.
* Add initial test-bootstrap method.

## `promisify` decorator
`promisify` decorator had been renamed to `exported`
Also add logic to handle methods which do not return `IFuture`

Modify README.md according to latest changes.

### CODE REVIEW

File: decorators.ts [hunk 1]
```diff
@@ -4,17 +4,22 @@
...
 		$injector.publicApi.__modules__[moduleName] = $injector.publicApi.__modules__[moduleName] || {};
 		$injector.publicApi.__modules__[moduleName][propertyKey] =  (...args: any[]): any => {
 				return new Promise(function(resolve: any, reject: any) {
 						let originalModule = $injector.resolve(moduleName);
-						fiberBootstrap.run(function () {
-							// TODO: Handle cases when function does not return IFuture
-							// TODO: Handle cases when function throws error
-							resolve(originalModule[propertyKey].apply(originalModule, args).wait());
-						});
+						let originalMethod: any = originalModule[propertyKey];
```
Comment (teobugslayer) [inline decorators.ts]: any -> Function

File: decorators.ts [hunk 2]
```diff
@@ -4,17 +4,22 @@
...
+						let result = originalMethod.apply(originalModule, args)
+						if(result && typeof result.wait === "function") {
+							fiberBootstrap.run(function () {
+								// TODO: Handle cases when function throws error
```
Comment (teobugslayer) [inline decorators.ts]: Can we add the try/catch in this PR?

Comment (rosen-vladimirov) [inline decorators.ts]: of course we can :) I'll add it

**Comment by ligaz** [comment]
Can you experiment with [`tsd`](http://definitelytyped.org/tsd/) for managing TypeScript declaration files? This will allow us not to commit them in version control.

**Comment by rosen-vladimirov** [comment]
@ligaz  I'll check `tsd` and if I succeed, I'll change our code in a separate PR.
I'm closing this one as the changes are included in this one: https://github.com/telerik/mobile-cli-lib/pull/410 All fixes will be applied there

**Pull request closed without merging**